Protos/api.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package api;
+
+service SchedulerApi {
+    rpc SubmitTask(TaskRequest) returns (TaskResponse) {}
+    rpc TaskCompleted(TaskCompletedRequest) returns (TaskCompletedResponse) {}
+    rpc RegisterWorker(RegisterWorkerRequest) returns (RegisterWorkerResponse) {}
+}
+
+service WorkerApi {
+    rpc GetResult(GetResultRequest) returns (GetResultResponse) {}
+    rpc ExecuteTask(TaskRequest) returns (TaskResponse) {}
```

Comment:
we would need another API to get task from scheduler?

Reply:
Sure, but do we need this for the initial flow?

Reply:
We are going into the model where scheduler sends the requests to worker right?
So there would be no API to get task from scheduler. 

Though, there should be an API in workerAPI to SubmitTask (to be used by scheduler) - left a comment for it

Reply:
Renamed to SubmitTask in the latest commit.